Trees/DepthFirstSearch.js

```diff
@@ -1,76 +1,213 @@
 /*
- * Author: Surendra Kumar
+ * Author: Sundaram Kumar Jha
  * DFS Algorithm implementation in JavaScript
  * DFS Algorithm for traversing or searching graph data structures.
 */
 
-function traverseDFS (root) {
-  const stack = [root]
-  const res = []
+function dfs(tree) {
+  if (!tree) return;
+  console.log(tree.value);
```

Comment:
This should either be calling a visitor function or yielding. No side effects.

Reply:
But for what

Reply:
For reusability.